be/src/exec/json_scanner.cpp

```diff
@@ -493,31 +500,76 @@ Status JsonReader::_read_rows(Chunk* chunk, int32_t rows_to_read, int32_t* rows_
 }
 
 Status JsonReader::_construct_row_without_jsonpath(simdjson::ondemand::object* row, Chunk* chunk) {
-    std::unordered_map<std::string_view, SlotDescriptor*> slot_desc_dict(_slot_desc_dict);
+    _parsed_columns.assign(chunk->num_columns(), false);
 
     try {
-        std::ostringstream oss;
+        uint32_t key_index = 0;
         for (auto field : *row) {
+            int column_index;
             std::string_view key = field.unescaped_key();
 
-            // look up key in the slot dict.
-            auto itr = slot_desc_dict.find(key);
-            if (itr == slot_desc_dict.end()) {
-                continue;
-            }
+            // _prev_parsed_position records the chunk column index for each key of previous parsed json object.
+            // For example, if previous json object is
+            // {
+            //      'a':1,
+            //      'b':2,
+            // }
+            // and key 'a' refers to the 1st column of chunk, key 'b' refers to the 2nd column of chunk,
+            // then _prev_parsed_position is [ {'a', 1, int}, {'b', 2, int}].
+            // At this time, suppose the next parsed json object is
+            // {
+            //      'a':10,
+            //      'b':15,
+            //      'c':25
+            // }
+            // through the _prev_parsed_position, we can know that the column index for 'a' is 1, and the column
+            // index for 'b' is 2. Since previous parsed json object doesn't contain 'c', key 'c' 's column index
+            // needs to be searched from the _slot_desc_dict, and if the key 'c' refers to the 3rd column of chunk,
+            // then we will update the _prev_parsed_position to be [{'a', 1, int}, {'b', 2, int}, {'c', 3, int}].
+            if (_prev_parsed_position.size() > key_index && _prev_parsed_position[key_index].key == key) {
+                // obtain column_index from previous parsed position
+                column_index = _prev_parsed_position[key_index].column_index;
```

Comment:
Why not to update the column_index here?

Reply:
because parsed key's column_index of the current json object is same as previous json object, no need to update column_index.